contrib/pyln-client/pyln/client/lightning.py

```diff
@@ -88,9 +88,10 @@ def __repr__(self) -> str:
 
     def to_satoshi(self) -> Decimal:
         """
-        Return a Decimal representing the number of satoshis.
+        Return a Decimal representing the number of satoshis,
+        rounded up to the nearest satoshi
         """
-        return Decimal(self.millisatoshis) / 1000
+        return (Decimal(self.millisatoshis) + 999) // 1000
```

Comment:
```python
import math
Decimal(math.ceil(self.millisatoshi / 1000))
```

Reply:
I'm not sure that we should just be rounding up or down on a whim, we need to keep the `Millisatoshi` as precise as possible, and then let the final step in the computation specify whether we want to round up or down.

Any attempt to do so implicitly will likely cause issues because what might seem intuitive now, may not be when looked at from another side.

Reply:
Imo it's a mistake to return a non-rounded response from `to_satoshi` -- a partial value satoshi isn't *correct*. Taking a millisatoshi to a satoshi *should* be the last step in any millisatoshi operations?

Reply:
Maybe the better solution here is to change the interface to accept an optional `whole` argument.

e.g.
```
def to_satoshi(self, whole=False)
```

Reply:
Agreed, however you're also inverting the semantics of the rounding of an existing API, which may break callers.

I was thinking more of deprecating `to_satoshi` as is and then add `floor()` and `ceil()` methods to the `Millisatoshi` class that returns the msat amount rounded in the desired direction.

Reply:
I think if `to_satoshi` returns a `Decimal` we should not round internally at all, and defer the rounding to the caller.

Reply:
You're right, modifying `to_satoshi` is the wrong approach, especially given that it does return the `Decmial` type.

Added a new method `to_whole_satoshi`, which returns an `int` type.

Reply:
Fwiw I wasn't a hugely into adding a call to `import math`, but if you think it's better than the inline math version I'd be happy to update/change it.